Add sponsorship requirements to the technical committee charter. (#2698)
open-telemetry/community: tech-committee-charter.md

The TC and entire technical community will follow any processes as may be specified by the Cloud Native Computing Foundation relating to the intake and license compliance review of contributions, including the CNCF IP Policy.

### Sponsorship Requirements

The TC will establish and maintain sponsorship across the SIGs and subprojects of OpenTelemetry.

> SIGs will be classified into three classifications of Sponsorship requirement:

Reviewer: It may be useful to include examples of such existing SIGs in the PR description to make the classification easier to understand.

Author: We discussed this in both the GC + TC meetings today.  PTAL at meeting notes and the video recordings.

I agree examples are useful.  I think that's an exercise we'll do over the next week, but I don't think it belongs in the PR.  I'll add the results and examples in the PR description.

Reviewer: >I don't think it belongs in the PR. I'll add the results and examples in the PR description.

I agree. That's what I meant: to add to the description, not to `tech-committee-charter.md`.

> * *Escalation Sponsorship*: The TC sponsor will be a primary means of escalation of cross-project concerns from the SiG. This communication is primarily

Reviewer 2: ```suggestion
* *Escalation Sponsorship*: The TC sponsor will be a primary means of escalation of cross-project concerns from the SIG. This communication is primarily
```

> * *Escalation Sponsorship*: The TC sponsor will be a primary means of escalation of cross-project concerns from the SiG. This communication is primarily

Reviewer 3: SIGs that are created as a result of a project (regardless if they plan to be "permanent" SIGs or dissolved after the project finishes) require a sponsor. Currently, there is a clause that allows TC to delegate sponsorship to others. Would this change involve a corresponding change to project management to specify that when a project sponsorship is delegated, there should still be a TC in the escalation path?

It almost feels like a project should always have a guiding/leading sponsor, which may be a TC member, or someone delegated by a TC member who would then act as an escalation sponsor for that SIG.

Author: > Would this change involve a corresponding change to project management to specify that when a project sponsorship is delegated, there should still be a TC in the escalation path?

I'll clarify this - Every SIG should have, at a  minimum, escalation sponsorship.

Additionally (and I will add this to the text), you can use a "higher level" sponsorship to satisfy a lower level requirement.  So a "Leading" Sponsorhip role would satisfy a "Escalation" sponsorship requirement for a SIG.

Author: Ok, updated the text to denote this, PTAL

>   offline, e.g. via CNCF slack or issue communication. The TC sponsor will check-in with SIG leadership quarterly.  Additionally, the TC will aide the SIG

Reviewer 4: I'm not sure quarterly is often enough. I'd argue for monthly if possible. A check-in shouldn't take longer than a few minutes.

check-in is not really defined here. Is that intentional? Is the TC ensuring the SIG is fulfilling their charter, or are they helping unblock issues, or are they simply ensuring the SIG is "healthy" and making progress? I assume it's some combination of the above but I'd define what the expectations are.

Aside: this is one of those moments (which come up very often) where it would be _very_ helpful if every SIG had a charter (which the community repo documentation says they're supposed to have) with explicit goals and an end/renewal date.

>   offline, e.g. via CNCF slack or issue communication. The TC sponsor will check-in with SIG leadership quarterly.  Additionally, the TC will aide the SIG

Author: Updated this to remove the time denotation completely and delegate to the `gc-check-ins.md` description.  This explicitly shows how it will now be a combined check-in and leaves the cadence open for us to sort out later.

  in developing quarterly road maps during this check-in.

> * *Guiding Sponsorship*: The TC sponsor is an active participant of the SIG. The sponsor will help direct the SIGs efforts to match the technical goals of OpenTelemetry.

Reviewer 4: Are guide responsibilities _in addition to_ escalation responsibilities, or would there be a separate TC member who is the escalation assignee? What if there are multiple TC members active in a SIG?

Reviewer 5: This is stated below:

> Higher level sponsorship automatically fulfills lower level sponsorship requirements

Maybe that should be above the list already to avoid confusion, because I had the same question initially.

> * *Leading Sponsorship*: The TC sponsor is an active member of SIG leadership, driving the direct success of the project or proposal.

Reviewer 4: I understand from the name what this is, but the description is essentially the same as the guiding. I'd expand this to be explicit about the TC member being responsible for the SIG's completion of their charter goals. I'd also make it a requirement to at least attend SIG meetings, and run them in the absence of another maintainer.

Reviewer 4: I just thought of one more point related to this. Is every SIG expected to have a TC leading sponsor, or is a spec sponsor enough to cover this role? I think any SIG that has a non-TC spec sponsor leader should at least have an escalation TC sponsor.

Author: Updated - Every SIG is required to at least have "escalation" sponsorship.

I like the language your'e using around Leading = owning outcomes.  I'll take a crack at updating the description shortly.

Author: Ok, fully updated, PTAL

As part of their active leadership, every TC member will be required to sponsor SIGs across OpenTelemetry.

> Every TC member is required to provide a  minimum of one (1) *Escalation* sponsorship. All SIGs requiring escalation sponsorship will be divided across the TC.

Reviewer 2: Which SIGs require escalation sponsorship? Which SIGs require guiding / leading sponsorship? And how many sponsors?

Author: Great question.  As discussed in the SIG - Let's put together a strawman for this so we're comfortable with the numbers in this PR and then we can formalize sponsorship requirements across SIGs in follow up work.

> Every TC member is required to provide a  minimum of one (1) *Escalation* sponsorship. All SIGs requiring escalation sponsorship will be divided across the TC.

Reviewer 6: ```suggestion
Every TC member is required to provide a minimum of one (1) *Escalation* sponsorship. All SIGs requiring escalation sponsorship will be divided across the TC.
```

Every TC member is required to provide a minimum of two (2) *Guiding* or *Leading* sponsorships.

> Every TC member is limited to a maximum of two (2) *leading* sponsorships. This is to keep OpenTelemetry efforts focused and reduce fragmented attention on

Reviewer 6: ```suggestion
Every TC member is limited to a maximum of two (2) *Leading* sponsorships. This is to keep OpenTelemetry efforts focused and reduce fragmented attention on
```

> core issues.

Author: Added a clause for this.

> A TC member may have a time where they are not meeting the minimum sponsorship requirement, e.g. when major efforts have acheived results, and new projects are being formulated.

Reviewer 4: May also want to consider a "time off policy" for extended vacations and such. Another TC member presumably would be expected to cover duties for some period of time?

The TC and entire technical community will follow any processes as may be specified by the Cloud Native Computing Foundation relating to the intake and license compliance review of contributions, including the CNCF IP Policy.

### Sponsorship Requirements

The TC will establish and maintain sponsorship across the SIGs and subprojects of OpenTelemetry.

SIGs will be classified into three classifications of Sponsorship requirement:

* *Escalation Sponsorship*: The TC sponsor will be a primary means of escalation of cross-project concerns from the SIG. This communication is primarily

>   offline, e.g. via CNCF slack or issue communication. The TC sponsor will participate in [GC check-ins](gc-check-ins.md) with SIG leadership.

Reviewer 6: ```suggestion
  offline, e.g. via CNCF Slack or issue communication. The TC sponsor will participate in [GC check-ins](gc-check-ins.md) with SIG leadership.
```

>   Additionally, the TC will aide the SIG in developing roadmaps during this check-in, and aligning to overall OpenTelemetry goals.

Reviewer 7: ```suggestion
  Additionally, the TC will aid the SIG in developing roadmaps during this check-in, and aligning to overall OpenTelemetry goals.
```
"the TC will aid" or "the TC sponsor will aid"

>   Additionally, the TC will aide the SIG in developing road maps during this check-in, and aligning to overall OpenTelemetry goals.

Reviewer 6: ```suggestion
  Additionally, the TC will aid the SIG in developing roadmaps during this check-in, and aligning to overall OpenTelemetry goals.
```

Reviewer 6: You decide which one is better since your English is way better than me (English is my 2nd language). I asked Google AI and got this suggestion:

![image](https://github.com/user-attachments/assets/88f1d93c-5381-4fee-a672-bcd3227747e4)

Author: I just meant I had it as `roadmap` and the build failed.  It's not MY choice, it's our spellcheck tool.

Reviewer 5: I am surprised that the spell checker has issues with "roadmap", but if this persists, then you can add it to the wordlist and no longer worry about it

Author: @svrnm Where is the wordlist?  I'll update it there.

Reviewer 5: https://github.com/open-telemetry/community/blob/main/.cspell.yaml

Author: updated the dictionary and use roadmaps now.  Thanks @svrnm !

* *Guiding Sponsorship*: The TC sponsor is an active participant of the SIG. The sponsor will help direct the SIGs efforts to match the technical goals of OpenTelemetry.
* *Leading Sponsorship*: The TC sponsor is an active member of SIG leadership, driving the direct success of the project or proposal.

All SIGs within OpenTelemetry will be required to have some level of sponsorship requirement (at least *Escalation*). Higher level sponsorship automatically fulfills
lower level sponsorship requirements, e.g. *Guiding* sponsorship fulfills an *Escalation* sponsorship requirement, as *Leading* sponsorship would fulfill an
*Escalation* sponsorship requirement. Sponsorship is not exclusive. More than one TC member may sponsor a SIG.

As part of their active leadership, every TC member will be required to sponsor SIGs across OpenTelemetry.

> Every TC member is required to provide a minimum of one (1) *Escalation* sponsorship. All SIGs requiring escalation sponsorship will be divided across the TC.

Reviewer 5: ```suggestion
All SIGs requiring escalation sponsorship will be divided across the TC. Every TC member is required to provide a minimum of one (1) *Escalation* sponsorship. 
```

I would flip those two sentences, because the latter is the important one, the other one just clarifies that there should be at least 1 (in a worst case there are `n-1` TC members with 1 escalation sponsorship, and `1` TC member doing all the other ones, but I assume the TC wants to keep the flexibility to distribute load on their own)

> Every TC member is required to provide a minimum of two (2) *Guiding* or *Leading* sponsorships.

Reviewer 5: To clarify that, this would also allow a TC member to only be Guiding Sponsor for 2 projects, and not leading a single one?

Author: Yes.

Every TC member is limited to a maximum of two (2) *Leading* sponsorships. This is to keep OpenTelemetry efforts focused and reduce fragmented attention on
core issues.

A TC member may have a time where they are not meeting the minimum sponsorship requirement, e.g. when major efforts have achieved results, and new projects are being formulated.

> This time should not exceed a six month period.

Reviewer 5: What if this time is exceeded? Is there any consequence?

Author: Yes - I think the GC takes action.  We don't describe the consequence for meeting attendance here either, but I believe the GC is the enforcer in both of those situations.

  offline, e.g. via CNCF Slack or issue communication. The TC sponsor will participate in [GC check-ins](gc-check-ins.md) with SIG leadership.
  Additionally, the TC sponsor will aid the SIG in developing roadmaps during this check-in, and aligning to overall OpenTelemetry goals.
* *Guiding Sponsorship*: The TC sponsor is an active participant of the SIG. The sponsor will help direct the SIGs efforts to match the technical goals of OpenTelemetry.
* *Leading Sponsorship*: The TC sponsor is an active member of SIG leadership, driving the direct success of the project or proposal.

All SIGs within OpenTelemetry will be required to have some level of sponsorship requirement (at least *Escalation*). Higher level sponsorship automatically fulfills
lower level sponsorship requirements, e.g. *Guiding* sponsorship fulfills an *Escalation* sponsorship requirement, as *Leading* sponsorship would fulfill an
*Escalation* sponsorship requirement. Sponsorship is not exclusive. More than one TC member may sponsor a SIG.

As part of their active leadership, every TC member will be required to sponsor SIGs across OpenTelemetry.

> All SIGs requiring escalation sponsorship will be divided across the TC. Every TC member is required to provide a minimum of one (1) *Escalation* sponsorship. 

Reviewer 8: is this right?

> Every TC member is required to provide a minimum of one (1) *Escalation* sponsorship.

Reviewer 9: The following paragraph contradicts it, so we'll need to update one of these

Author: Fixed, my bad.
